src/Microsoft.SqlTools.ServiceLayer/Migration/Models/StaticTokenCredential.cs

```diff
@@ -0,0 +1,35 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Microsoft.SqlTools.ServiceLayer.Migration.Models
+{
+    /// <summary>
+    /// Temp token creadentials to interact with ArmClient class. 
+    /// The token passed to this class should be a newly request token, because this class doesn't renew the token.
+    /// Once MSAL is rolled out on ADS, we will implement a way to use the same ADS token cache configured by ADS. 
```

Comment:
the Azure Core extension apparently recently migrated to using MSAL instead of ADAL - not sure if this opens the door for us to start using MSAL as well

Reply:
Yes, we'll use the same configuration to use information stored in the cache and replace the token parameter for the tenant.